### PR TITLE
feat: copy with row and col

### PR DESCRIPTION
**Pull request by YardWill**
### 👀 PR includes

✨ Feature

- [x] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve performance.
- [ ] Type optimization

🐛 Bugfix

- [ ] Bug fix issue #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Other

### 📝 Description

suppport in table sheet
1. col select
2. row select
3. all select

### 🖼️ Screenshot

|  Before  |  After  |
|----|----|
|  ❌  |  ✅  |

### CODE REVIEW

File: packages/s2-core/src/utils/export/copy.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { copyToClipboard } from '.';
-import { S2CellType } from '@/common/interface';
+import { S2CellType, ViewMeta } from '@/common/interface';
 import { SpreadSheet } from '@/sheet-type';
+import { InteractionStateName } from '@/.';
```
Comment (xingwanying) [inline packages/s2-core/src/utils/export/copy.ts:4]:
```suggestion
import { InteractionStateName } from '@/common/constant/interaction';
```

**Review by xingwanying — COMMENTED** [review]
后面辛苦大佬补一波单测嗷～贴个动图秀下效果？

File: packages/s2-core/src/utils/export/copy.ts (hunk 1 above)
Comment (YardWill) [reply packages/s2-core/src/utils/export/copy.ts]: 加了测试 动图你们都用什么工具搞的？

File: packages/s2-core/__tests__/unit/utils/copy-spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      .filter((e) => e.cellType === CellTypes.ROW_CELL)[3];
+
+    s2.interaction.changeState({
+      cells: [getCellMeta(cell)],
+      stateName: InteractionStateName.SELECTED,
+    });
+    expect(getSelectedData(s2).split('\t').length).toBe(5);
+  });
+
+  it('copy all data', () => {
```
Comment (lijinke666) [inline packages/s2-core/__tests__/unit/utils/copy-spec.ts:68]: 单测名统一加一个 should 吧？  should copy all data

File: packages/s2-core/src/utils/export/copy.ts [hunk 3]
```diff
@@ -47,9 +48,65 @@ export const getTwoDimData = (cells: S2CellType[]) => {
...
+    .join('\n');
+};
+
 export const getSelectedData = (spreadsheet: SpreadSheet) => {
-  const cells = spreadsheet.interaction.getActiveCells();
-  const data = processCopyData(getTwoDimData(cells));
+  const interaction = spreadsheet.interaction;
+  const cells = interaction.getActiveCells();
+
+  let data;
```
Comment (lijinke666) [inline packages/s2-core/src/utils/export/copy.ts:96]: let data: string

File: packages/s2-core/src/utils/export/copy.ts [hunk 4]
```diff
@@ -47,9 +48,65 @@ export const getTwoDimData = (cells: S2CellType[]) => {
...
+  const selectedCols = cells.filter((e) => e.cellType === 'colCell');
+  const selectedRows = cells.filter((e) => e.cellType === 'rowCell');
```
Comment (lijinke666) [inline packages/s2-core/src/utils/export/copy.ts:98]: 不要取 e 这种无意义的形参， 要么  cell, 要么 解构, ({cellType})

File: packages/s2-core/src/utils/export/copy.ts [hunk 5]
```diff
@@ -47,9 +48,65 @@ export const getTwoDimData = (cells: S2CellType[]) => {
...
+};
+
 export const getSelectedData = (spreadsheet: SpreadSheet) => {
-  const cells = spreadsheet.interaction.getActiveCells();
-  const data = processCopyData(getTwoDimData(cells));
+  const interaction = spreadsheet.interaction;
+  const cells = interaction.getActiveCells();
+
+  let data;
+  const selectedCols = cells.filter((e) => e.cellType === 'colCell');
```
Comment (lijinke666) [inline packages/s2-core/src/utils/export/copy.ts:97]: colCell => CellTypes.ColCell

File: packages/s2-core/src/utils/export/copy.ts (hunk 1 above)
Comment (xingwanying) [reply packages/s2-core/src/utils/export/copy.ts]: > 加了测试 动图你们都用什么工具搞的？

Kap

**Review by xingwanying — DISMISSED** [review]
解决下上面几个comments即可

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**